### ISSUE TITLE
Cluster in not-ready state after fixing a wrong Gitrepo URL

### ISSUE THREAD

**Issue by mmartin24** [rancher/dashboard#11404]
## Issue
While checking other issue, it was noticed that when fixing a wrong URL from a Gitrepo, the cluster remains  in `NotReady` state

### Steps to reproduce:
- Deploy Rancher 2.9 
- Deploy a Girepo that will produce an error, for instance with a space:
```
Repository URL: `https://github.com/rancher/fleet-test-data ` (notice the space at the end)
Branch: `master`
Paths: `qa-test-apps/nginx-app`
```
- Wait until fails
- Fix it by removing the space
- Wait until is corrected
- Go to `Continuous Delivery` > `Clusters`

### Observed behavior
Cluster in `Not Ready` status for a long, long time

![Image](https://github.com/rancher/fleet/assets/37271841/9f5aa6c5-048c-4d6e-a06d-3d1c9b886932)

https://github.com/rancher/fleet/assets/37271841/5775df26-9134-4377-b997-8dbe518eb38b

---
### Test environment:
Single cluster Rancher: 	v2.9-1d34dbcbf674228c8e63d10449c6da35d8d16692-head
Fleet version:  `fleet:104.0.0+up0.10.0-rc.19`
K8s: `v1.28.8 +k3s1`

**Comment by mmartin24** [comment]
This seems fixed after 2.9.2:

[Screencast from 11-10-24 16:57:18.webm](https://github.com/user-attachments/assets/b79c5dfc-c992-408d-bb5a-c09e6b0a32a6)

Maybe the fix on this other issue helped: https://github.com/rancher/fleet/issues/2699

**Comment by gaktive** [comment]
@nwmac this looks done in 2.9.2 already. Should we update the milestone to reflect this accurately? or is there anything left for us to do?
